Added new employability pages covering competency interviews and intr… (#1722)
CodeYourFuture/curriculum: org-cyf-guides/content/employability/interview/introducing-yourself/index.md

---
title: Introducing Yourself
description:
emoji: 🖥️
weight: 6
---

Many hiring managers claim that they know whether a candidate is a good fit for a role within the first 5 minutes of an interview. It takes me a little longer than that, but more than one candidate has run aground trying to navigate the first question I ask in every single interview I conduct.

*Can you tell me a little about yourself?*

Many simple things conceal hidden depths, and judging by the occasional startled, or even panicked expressions of the candidates I have interviewed, this question is no different.

> Before we discuss what constitutes a "good" answer to this question, let's consider why I ask it in every interview — from interns to senior engineering leaders.

Reviewer: ```suggestion
Before we discuss what constitutes a "good" answer to this question, let's consider why it iis asked in every interview — from interns to senior engineering leaders.
```

I hope you can see just how powerful a tool this simple question can be. So why does it go wrong so often? Why is this question, more than any other, the one that sinks a candidate's chances? There are three fatal mistakes/problems that I've seen come up time and again.

## Three fatal mistakes

### 1. The all-too-brief answer

Whether as a result of no preparation, a nerves-induced brain freeze, or just poor communication skills, many unsuccessful candidates have attempted to answer this question in the form of a tweet — and a pre-2017 140 character tweet at that.

*"I'm Jack, I'm a software engineer and I'm here to interview for the role of senior software engineer with your company"* (118 chars) is boring, gives me nothing I can follow up on and is all information I already know. As I outline above — this is an important question. Give it some effort.

### 2. The NeverEnding Story answer

> This is far less common than a too-brief answer, but it has happened from time-to-time. I had one recent interview where the candidate spent 35 minutes answering this question — on a 45 minute call. He detailed every role he'd ever had and every project he'd ever been involved in. My co-interviewer faked a schedule clash to justify dropping off the call halfway through — if only I'd thought of that first. Be selective in what you choose to highlight.

Reviewer: ```suggestion
This is far less common than a too-brief answer, but it has happened from time-to-time. One of our volunteers had a recent interview where the candidate spent 35 minutes answering this question — on a 45 minute call. He detailed every role he'd ever had and every project he'd ever been involved in. Be selective in what you choose to highlight.
```

### 3. The incoherent answer

> This interview question offers insight into how the candidate communicates, for good and for bad. Modern software development is as much about good communication as it is about programming language syntax and design patterns — more so, even. When a candidate rambles aimlessly, hops from point to point, abandons sentences mid-way through, or is generally unintelligible, that's a bad start to an interview. And ultimately, that's how I will assume you're going to communicate whenever I ask for a project status update or how your weekend was.

Reviewer: ```suggestion
This interview question offers insight into how the candidate communicates, for good and for bad. Modern software development is as much about good communication as it is about programming language syntax and design patterns — more so, even. When a candidate rambles aimlessly, hops from point to point, abandons sentences mid-way through, or is generally unintelligible, that's a bad start to an interview. And ultimately, that's how many interviewers will assume you're going to communicate whenever you are asked for a project status update or how your weekend was.
```

## Guidelines for your answer

**Length:** 1-2 minutes is a nice length, but you can go up to 3 or 4 if you have enough relevant things to talk about — for example, a long career that you want to summarise.

**Topics to touch on:**

- Your experience, e.g. *"I've got x years of experience in backend development"*
- Your most recent role, e.g. *"I've been doing cool stuff at this company for the last x years"*
- Why you are interested in this role, e.g. *"I'm currently looking for a new role that will offer me opportunities to develop in areas x, y, and z"*

**Topics to avoid:**

- Criticism of yourself — this isn't the time to tell me you suck at paperwork (true story)
- Inappropriate personal information — I don't need to know that you have a bad relationship with your family (true story)
- Badmouthing current/past employer — don't do this; no matter how eloquently you describe your past employer's faults, there's a little voice in the back of my head wondering if you were the problem (many true stories)

> 

Reviewer 2: Another answer that puts me off that I've seen a lot here is if a candidate goes straight in for the hard sell, "I am Natalie and I'd be great for this job because of X,Y,Z". No harm in highlighting all your great skills and experience but I didn't ask "Why should we hire you".

## Examples

> So that's all theory — what might this answer look like in practice? Below I've drafted a few brief (but not too brief) answers to this question that I might have asked at various points in my career.

Reviewer: ```suggestion
So that's all theory — what might this answer look like in practice? Below are a few brief (but not too brief) answers to this question that this volunteer might have asked at various points in my career.
```

### Current career James

I'm James and I'm an experienced engineer and engineering leader with 17 years of experience across the public, private and academic sectors. My most recent role has been as a CTO of an ad-tech startup, where I lead a team of X engineers. Our software has been rolled out worldwide to Xm people and has been trusted by major brands such as BigCo and BiggerCo, but my wider career has seen me touch on a number of individual contributor and management roles, across software development, data science and artificial intelligence research. I'm currently looking for a role where someone will pay me to sit on a beach and drink cocktails (call me!).

---

> I hope this has been useful. It might seem like a lot of effort for what is, after all, just one of many questions you will be facing in your interview. But remember: **you only get one shot at a first impression. Make it a good one.**

Reviewer: ```suggestion
It might seem like a lot of effort for what is, after all, just one of many questions you will be facing in your interview. But remember: **you only get one shot at a first impression. Make it a good one.**
```
